src/main/scala/com/bazaarvoice/sswf/service/StepDecisionWorker.scala

```diff
@@ -0,0 +1,167 @@
+   */
+  @Nullable
+  def pollForDecisionsToMake(): DecisionTask = {
+    val decisionTask: DecisionTask = swf.pollForDecisionTask(new PollForDecisionTaskRequest().withDomain(domain).withTaskList(new TaskList().withName(taskList)).withIdentity(identity))
+    if (decisionTask.getTaskToken != null) {
+      decisionTask
+    } else {
+      null
+    }
+  }
+
+  /**
+   * Given the context of a workflow execution, decide what needs to be done next.
+   * @param decisionTask The task from SWF. Must not be null!
+   * @throws Throwable A variety of exceptions are possible. Make sure your worker threads can't die without replacement.
+   * @return The decision that we sent to SWF
+   */
+  def makeDecision(@NotNull decisionTask: DecisionTask): RespondDecisionTaskCompletedRequest = {
+    require(decisionTask != null, "decisionTask must not be null")
+
+    val (newDecisionTask, events) = getFullHistory(decisionTask)
+    val completedRequest: RespondDecisionTaskCompletedRequest = innerMakeDecision(newDecisionTask, events)
+
+    swf.respondDecisionTaskCompleted(completedRequest)
+
+    completedRequest
+  }
+
+  private[this] def innerMakeDecision(decisionTask: DecisionTask, events: List[HistoryEvent]): RespondDecisionTaskCompletedRequest = {
+    val history: StepsHistory[SSWFInput, StepEnum] = HistoryFactory.from(events, inputParser)
+
+    val input = history.input
+
+    def respond(d: Decision): RespondDecisionTaskCompletedRequest =
+      new RespondDecisionTaskCompletedRequest().withDecisions(List(d)).withTaskToken(decisionTask.getTaskToken)
+
+    def schedule(activity: Option[StepEnum]) = activity match {
+      case None       =>
+        val message = "No more activities to schedule."
+        //        LOG.info("Wf complete: %s".format(message))
+        workflowDefinition.onFinish(input, history, message)
+        var attributes: CompleteWorkflowExecutionDecisionAttributes = new CompleteWorkflowExecutionDecisionAttributes
+        attributes = attributes.withResult(message)
+
+        new Decision().withDecisionType(DecisionType.CompleteWorkflowExecution).withCompleteWorkflowExecutionDecisionAttributes(attributes)
+      case Some(step) =>
+        //        LOG.info(s"scheduling $stage")
+        val scheduleActivityTaskDecisionAttributes: ScheduleActivityTaskDecisionAttributes = new ScheduleActivityTaskDecisionAttributes()
+           .withActivityId(step.name)
+           .withActivityType(new ActivityType().withName(step.name).withVersion(step.version))
+           .withHeartbeatTimeout("NONE")
+           .withTaskList(new TaskList().withName(taskList))
+           .withInput(inputParser.serialize(input))
+
+        new Decision()
+           .withDecisionType(DecisionType.ScheduleActivityTask)
+           .withScheduleActivityTaskDecisionAttributes(scheduleActivityTaskDecisionAttributes)
+    }
+
+    def fail(shortDescription: String, message: String) = {
+      require(shortDescription.length < 256)
```

Comment:
if the requirement fails, it throws an IllegalArgumentException, so it's more like guava's checkArgument.